src/trigger.c

```diff
@@ -79,34 +98,63 @@ trigger_create_on_chunk(Oid trigger_oid, char *chunk_schema_name, char *chunk_ta
 								 * twice */
 }
 
-
-
 void
-trigger_create_on_all_chunks(Hypertable *ht, Chunk *chunk)
+trigger_create_all_on_chunk(Hypertable *ht, Chunk *chunk)
 {
-	ScanKeyData skey;
-	Relation	tgrel;
-	SysScanDesc tgscan;
-	HeapTuple	htup;
+	Relation	htrel;
+
+	htrel = relation_open(ht->main_table_relid, AccessShareLock);
+
```

Comment:
Because this function creates chunk triggers in the loop, while `relation_has_transition_table_trigger` just checks for transition tables.

I could try to reduce code duplication by doing a loop that takes an optional handler function as arg.